src/Controls/ImmersiveControls.js

```diff
@@ -0,0 +1,109 @@
+import FirstPersonControls from 'Controls/FirstPersonControls';
+import AnimationPlayer, { AnimatedExpression } from 'Core/AnimationPlayer';
+
+// Expression used to damp camera's moves
+function moveCameraExp(root, progress) {
+    const dampingProgress = 1 - Math.pow((1 - (Math.sin((progress - 0.5) * Math.PI) * 0.5 + 0.5)), 2);
+    root.camera.position.lerpVectors(root.positionFrom, root.positionTo, dampingProgress);
+}
+/**
+ * @classdesc Camera controls adapted to immersive view.
+ * It allow the user to navigate threw {@link OrientedImageLayer} data.
+ *
+ * <ul> Bindings inherited from FirstPersonControls
+ * <li><b> up + down keys : </b> forward/backward </li>
+ * <li><b> left + right keys: </b> strafing movements </li>
+ * <li><b> pageUp + pageDown: </b> vertical movements </li>
+ * <li><b> mouse click+drag: </b> pitch and yaw movements (as looking at a panorama) </li>
+ * </ul>
+ * <ul> Bindings added
+ * <li><b> keys Z : </b> Move camera to the next panoramic </li>
+ * </ul>
+ * @extends FirstPersonControls
+ */
+class ImmersiveControls extends FirstPersonControls {
```

Comment:
It isn't necessary that this camera is specific to immersive. It should be a camera that follows a path. This path will consist of ordered positions. We could then use this camera for other user cases.

Reply:
I simplied this control, now it only store currentPosition and nextPosition , and it's the layer that is updating these twos current and next position.

Reply:
Could you rename and clean up class and properties. Could you remove `pano reference, this.layers`...